chart/templates/jobs/cache-refresh/job.yaml

```diff
@@ -0,0 +1,28 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+{{- if .Values.images.jobs.cacheRefresh }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  labels: {{ include "labels.cacheRefresh" . | nindent 4 }}
+  name: "{{ include "name" . }}-job-cache-refresh"
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-weight": "-1"
+    "helm.sh/hook-delete-policy": before-hook-creation
```

Comment:
I didn’t check the code, but it’s very important for this job to not be blocking if the DB is not yet created or ready, to avoid blocking all the deployment process during the first install 😉

Reply:
Ah, good point. The job works as the migration one: in case the connection to the database cannot be established, the script exits with code 1. See:
- the new job: https://github.com/huggingface/datasets-server/pull/912/files#diff-b1c58a99fef3df220ff883909890e39deb67ad586c6e0dfb833290e4a6d21d40
- the other job (migration) that we fixed to avoid blocking: https://github.com/huggingface/datasets-server/blob/main/jobs/mongodb_migration/src/mongodb_migration/main.py